src/commands/operation.rs

```diff
@@ -206,15 +206,19 @@ fn view_with_desired_portions_restored(
     new_view
 }
 
-fn process_what_arg(what_arg: &[UndoWhatToRestore]) -> BTreeSet<UndoWhatToRestore> {
+fn process_what_arg(
+    what_arg: &[UndoWhatToRestore],
+    colocated: bool,
+) -> BTreeSet<UndoWhatToRestore> {
     if !what_arg.is_empty() {
         what_arg.iter().cloned().collect()
     } else {
-        btreeset!(
-            UndoWhatToRestore::Repo,
-            UndoWhatToRestore::RemoteTracking,
-            UndoWhatToRestore::GitTracking
-        )
+        let mut default_what =
+            btreeset!(UndoWhatToRestore::Repo, UndoWhatToRestore::RemoteTracking);
+        if !colocated {
+            default_what.insert(UndoWhatToRestore::GitTracking);
+        }
+        default_what
```

Comment:
nit: better to add inline comment why not restoring `git_refs` is safe.
If we didn't have automated export, it would bring us to peculiar state.